src/Compilers/VisualBasic/Portable/Symbols/MissingMetadataTypeSymbol.vb

```diff
@@ -167,23 +172,50 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 End Get
             End Property
 
-            Public Overrides ReadOnly Property SpecialType As SpecialType
+            Private ReadOnly Property TypeId As Integer
                 Get
                     If _lazyTypeId = -1 Then
-                        Dim typeId As SpecialType = SpecialType.None
+                        Dim computedTypeId As SpecialType = SpecialType.None
                         Dim containingAssembly As AssemblySymbol = _containingModule.ContainingAssembly
 
                         If (Arity = 0 OrElse MangleName) AndAlso
                            containingAssembly IsNot Nothing AndAlso containingAssembly Is containingAssembly.CorLibrary AndAlso _containingModule.Ordinal = 0 Then
                             ' Check the name 
                             Dim emittedName As String = MetadataHelpers.BuildQualifiedName(_namespaceName, MetadataName)
-                            typeId = SpecialTypes.GetTypeFromMetadataName(emittedName)
+                            computedTypeId = SpecialTypes.GetTypeFromMetadataName(emittedName)
                         End If
 
-                        Interlocked.CompareExchange(_lazyTypeId, typeId, -1)
+                        Interlocked.CompareExchange(_lazyTypeId, computedTypeId, -1)
                     End If
 
-                    Return CType(_lazyTypeId, SpecialType)
+                    Return _lazyTypeId
+                End Get
+            End Property
+
+            Public Overrides ReadOnly Property SpecialType As SpecialType
+                Get
+                    Dim typeId As Integer = Me.TypeId()
+                    Return If(typeId >= WellKnownType.First, SpecialType.None, CType(_lazyTypeId, SpecialType))
+                End Get
+            End Property
+
+            Friend Overrides ReadOnly Property ErrorInfo As DiagnosticInfo
+                Get
+                    Select Case Me.TypeId()
+                        Case WellKnownType.System_ValueTuple_T1,
+                             WellKnownType.System_ValueTuple_T2,
+                             WellKnownType.System_ValueTuple_T3,
+                             WellKnownType.System_ValueTuple_T4,
+                             WellKnownType.System_ValueTuple_T5,
+                             WellKnownType.System_ValueTuple_T6,
+                             WellKnownType.System_ValueTuple_T7,
+                             WellKnownType.System_ValueTuple_TRest
+                            Return ErrorFactory.ErrorInfo(ERRID.ERR_ValueTupleTypeRefResolutionError, Me, ContainingModule.Name)
```

Comment:
Will do. I'll add two more errors.

One question on that though. `Errors.vb` has the following message, but I cannot find the class mentioned:
```VB
    '//If you make any change involving these errors, such as creating more specific versions for use
        '//in other contexts, please make sure to appropriately modify Bindable::ResolveOverloadingShouldSkipBadMember
        ERR_UnreferencedAssembly3 = 30652
        ERR_UnreferencedModule3 = 30653
```